### PR TITLE
Send thanks button in more details fragment

### LINKED ISSUE #5421: Button to thank the uploader, in Media Details
Unless the user is the uploader.

The code to thank the person who uploaded an image has been implemented already, in Peer Review, it can be reused.

The button could be similar to the Copy Wikitext button, and placed directly below it:

### PR DESCRIPTION
**Pull request by srishti-R**
**Description (required)**

Fixes #5421 

What changes did you make and why?
Send thank button will appear only when the image is not uploaded by author.
Screenshot_20240108-203814_Commons
Screenshot_20240108-203825_Commons
Screenshot_20240108-203826_Commons
Screenshot_20240108-203859_Commons

**Tests performed (required)**

Tested prodDebug on Samsung galaxy M31 with API level 31.

**Screenshots (for UI changes only)**

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -431,6 +447,12 @@ public void onResume() {
             enableProgressBar();
         }
 
+        if (AccountUtil.getUserName(getContext()) != null && media != null && AccountUtil.getUserName(getContext()).equals(media.getAuthor())) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:450]: How about splitting this very long line?

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 2]
```diff
@@ -780,12 +802,63 @@ public void onMediaDetailCoordinatesClicked(){
...
+            "[[" + media.getFilename() + "|thumb|" + media.getFallbackDescription() + "]]";
+        Utils.copy("wikiCode", data, getContext());
         Timber.d("Generated wikidata copy code: %s", data);
 
-        Toast.makeText(getContext(), getString(R.string.wikicode_copied), Toast.LENGTH_SHORT).show();
+        Toast.makeText(getContext(), getString(R.string.wikicode_copied), Toast.LENGTH_SHORT)
+            .show();
+    }
+
+    @OnClick(R.id.sendThanks)
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:815]: Would you mind writing some Javadoc for each added method?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -784,6 +784,8 @@ Upload your first media by tapping on the add button.</string>
   <string name="storage_permissions_denied">Storage Permissions Denied</string>
   <string name="unable_to_share_upload_item">Unable to share this item</string>
   <string name="permissions_are_required_for_functionality">Permissions are required for functionality</string>
+  <string name="send_thanks_to_author">Send thanks to author</string>
```
Comment (nicolas-raoul) [inline app/src/main/res/values/strings.xml:787]: How about a shorter "Thank the author"?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Working great!
I checked all places I could think of where images are visible, then checked that the thanks were correctly sent (at https://commons.wikimedia.org/wiki/Special:Log?type=thanks&user=Syced&page=&wpdate=&tagfilter=&wpFormIdentifier=logeventslist)

**Comment by nicolas-raoul** [comment]
In order to fix the unit tests, would you mind rebasing (or pulling) from main? Thanks :-)

**Comment by srishti-R** [comment]
@nicolas-raoul made the changes, sorry for the delay

**Review by nicolas-raoul — APPROVED** [review]
Sorry for the delay, it somehow escaped my attention.

It works great and the code looks good. Thanks a lot! :-)

**Pull request merged**